src/GuardClauses/GuardClauseExtensions.cs

```diff
@@ -126,12 +131,13 @@ public static string NullOrWhiteSpace([JetBrainsNotNull] this IGuardClause guard
         /// <param name="parameterName"></param>
         /// <param name="rangeFrom"></param>
         /// <param name="rangeTo"></param>
+        /// <param name="message">Optional. Custom error message</param>
         /// <returns><paramref name="input" /> if the value is inside the given range.</returns>
         /// <exception cref="ArgumentException"></exception>
         /// <exception cref="ArgumentOutOfRangeException"></exception>
-        public static int OutOfRange([JetBrainsNotNull] this IGuardClause guardClause, int input, [JetBrainsNotNull] string parameterName, int rangeFrom, int rangeTo)
+        public static int OutOfRange([JetBrainsNotNull] this IGuardClause guardClause, int input, [JetBrainsNotNull] string parameterName, int rangeFrom, int rangeTo, string? message = null)
         {
-            return OutOfRange<int>(guardClause, input, parameterName, rangeFrom, rangeTo);
+            return OutOfRange<int>(guardClause, input, parameterName, rangeFrom, rangeTo, message);
```

Comment:
I haven't checked the source for `OutOfRangeException` but it may have the same issue noted above for `ArgumentNullException`. In which case you'll need separate instantiation code for the appropriate overload of the constructor.

Reply:
The private method now throws a different exception as per overload. I assume that was what the next few comments pointed to?

```
if (string.IsNullOrEmpty(message))
{
    throw new ArgumentOutOfRangeException(parameterName, $"Input {parameterName} was out of range");
}
throw new ArgumentOutOfRangeException(message, (Exception?) null);
```

Reply:
Yes, as long as the existing behavior remains consistent when no message is provided - that's all I'm looking for. I don't want to break folks who might have behavior that depends on a specific error message. They shouldn't, especially in production, but they might have test assertions, etc. that look for the message, etc.